sql/core/src/main/scala/org/apache/spark/sql/execution/command/SetCommand.scala

```diff
@@ -172,7 +172,7 @@ object SetCommand {
 case class ResetCommand(config: Option[String]) extends RunnableCommand with IgnoreCachedData {
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    val defaults = sparkSession.sparkContext.conf
+    val defaults = sparkSession.sharedState.conf
```

Comment:
ah so this only adds the warehouse dir config compared to the `sparkContext.conf`?

Reply:
No. The `sharedState.conf` is a cloned `sparkContext.conf` with other initial options including the warehouse dir and other static/dynamic configs set by the 1st SparkSession instance which creates the one and only `sharedState`. 
If there is existing sc when we create the 1st SparkSession (e.g. see the PR description), the initial configs will go to the `sharedState.conf`, and we use this version of conf as defaults. Also, later created SparkSession with other options will not affect the `sharedState.conf`